src/model/resources/fusion/fusion.jl

```diff
@@ -0,0 +1,493 @@
+"""
+function ensure_fusion_expression_records!(dict::Dict)
+    for var in (FUSION_PARASITIC_POWER, FUSION_PULSE_START_POWER)
+        if var ∉ keys(dict)
+            dict[var] = Set{Symbol}()
+        end
+    end
+    var = FUSION_COMPONENT_ZONE
+    if var ∉ keys(dict)
+        dict[var] = Dict{Int, Set{AbstractString}}()
+    end
+end
+
+@doc raw"""
+    ensure_fusion_pulse_variable_records!(dict::Dict)
+
+    dict: a dictionary of model data
+
+    This should be called by each method that adds fusion formulations,
+    to ensure that certain entries in the model data dict exist.
+"""
+function ensure_fusion_pulse_variable_records!(dict::Dict)
+    for var in (FUSION_PULSE_START, FUSION_PULSE_UNDERWAY)
+        if var ∉ keys(dict)
+            dict[var] = Set{Symbol}()
+        end
+    end
+end
+
+@doc raw"""
+    fusion_parasitic_power_expressions(dict::Dict)
+
+    dict: a dictionary of model data
+
+    get listings of parasitic power expressions.
+    This is available only after a `fusion_formulation_*!` has been called.
+"""
+function fusion_parasitic_power_expressions(dict::Dict)::Set{Symbol}
+    dict[FUSION_PARASITIC_POWER]
+end
+
+@doc raw"""
+    fusion_parasitic_power_expressions(inputs::Dict, zone::Int)
+
+    inputs: model inputs
+
+    get listings of parasitic power expressions in a zone.
+    This is available only after a `fusion_formulation_*!` has been called.
+"""
+function fusion_parasitic_power_expressions(inputs::Dict, zone::Int)::Set{Symbol}
+    fusion_in_zone = inputs[FUSION_COMPONENT_ZONE][zone]
+    exprs = fusion_parasitic_total_name.(fusion_in_zone)
+    return Set(exprs)
+end
+
+@doc raw"""
+    fusion_pulse_start_expressions(dict::Dict)
+
+    dict: a dictionary of model data
+
+    get listings of pulse start expressions
+    This is available only after a `fusion_formulation_*!` has been called.
+"""
+function fusion_pulse_start_expressions(dict::Dict)::Set{Symbol}
+    dict[FUSION_PULSE_START]
+end
+
+function add_fusion_component_to_zone_listing(
+        inputs::Dict, r_id::Int, resource_component::AbstractString)
+    zone = inputs["R_ZONES"][r_id]
+    d = inputs[FUSION_COMPONENT_ZONE]
+    new_set = Set([resource_component])
+    if zone in keys(d)
+        union!(d[zone], new_set)
+    else
+        d[zone] = new_set
+    end
+end
+
+@doc raw"""
+    has_fusion(dict::Dict)::Bool
+
+    dict: a dictionary of model data
+
+    Checks whether the dictionary contains listings of fusion-related expressions.
+    This is true only after a `fusion_formulation_*!` has been called.
+"""
+function has_fusion(dict::Dict)::Bool
+    FUSION_PARASITIC_POWER in keys(dict)
+end
+
+#######################################
+# Define a data structure for a reactor
+#######################################
+
+# Base.@kwdef could be used if we enforce Julia >= 1.9
+# That would replace need for the keyword-argument constructor below
+Base.@kwdef struct FusionReactorData
```

Comment:
`Base.@kwdef` should be available in Julia versions 1.1 and later. However, it needs to be fully qualified because it wasn't exported until Julia 1.9.

Reply:
done.